rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportService.java

```diff
@@ -276,4 +297,164 @@ private Set<LicenseNameWithText> getExcludedLicenses(Set<String> excludedLicense
         return licenseInfoParsingResult.stream().map(LicenseInfoParsingResult::getLicenseInfo)
                 .flatMap(streamLicenseNameWithTexts).filter(filteredLicense).collect(Collectors.toSet());
     }
+    
+    public ByteBuffer getLicenseResourceBundleBuffer() throws TException {
+        return licenseClient.getLicenseReportDataStream();
+    }
+
+    public ByteBuffer downloadSourceCodeBundle(String projectId, HttpServletRequest request, User sw360User)
+            throws TException {
+        if (projectId==null || !validateProject(projectId, sw360User)) {
+            throw new TException("No project record found for the project Id : " + projectId);
+        }
+        Map<String, Set<String>> selectedReleaseAndAttachmentIds = getSelectedReleaseAndAttachmentIdsFromRequest(
+                request, false);
+        Set<String> selectedAttachmentIds = new HashSet<>();
+        selectedReleaseAndAttachmentIds.forEach((key, value) -> selectedAttachmentIds.addAll(value));
+        try {
+            Project project = projectclient.getProjectById(projectId, sw360User);
+            saveSourcePackageAttachmentUsages(project, sw360User, selectedReleaseAndAttachmentIds);
+            List<AttachmentContent> attachments = new ArrayList<>();
+            for (String id : selectedAttachmentIds) {
+                attachments.add(attachmentClient.getAttachmentContent(id));
+            }
+            return serveAttachmentBundle(attachments, request, project, sw360User);
+        } catch (TException e) {
+            throw new TException(e.getMessage());
+        }
```

Comment:
Rather than catching and throwing the same exception, remove the try/catch as exception handling is not done in this function.